### ISSUE TITLE
urlize_quoted_links got incorrect result when url path is not encoded with utf-8

### ISSUE THREAD

**Issue by ElnathMojo** [encode/django-rest-framework#6114]
## Description
When a url's path is not encoded with utf-8 such as EUC-JP, browsable api will return the url with incorrect link.
## Expected behavior
href value should be exactly same as the actual url.
```html
<a href="http://seesaawiki.jp/w/radioi_34/d/%b1%d2%b5%dc%a4%b5%a4%f3%a4%c1%a4%ce%ba%a3%c6%fc%a4%ce%a4%b4%a4%cf%a4%f3" rel="nofollow"><span class="str">http://seesaawiki.jp/w/radioi_34/d/%b1%d2%b5%dc%a4%b5%a4%f3%a4%c1%a4%ce%ba%a3%c6%fc%a4%ce%a4%b4%a4%cf%a4%f3</span></a>
```
## Actual behavior
urlize_quoted_links use django's smart_urlquote to quote the url. But smart_urlquote is using default utf-8 encoding and got incorrect result.
```html
<a href="http://seesaawiki.jp/w/radioi_34/d/%EF%BF%BD%D2%B5%DC%A4%EF%BF%BD%EF%BF%BD%EF%BF%BD%EF%BF%BD%EF%BF%BD%CE%BA%EF%BF%BD%EF%BF%BD%EF%BF%BD%EF%BF%BD%CE%A4%EF%BF%BD%EF%BF%BD%CF%A4%EF%BF%BD" rel="nofollow"><span class="str">http://seesaawiki.jp/w/radioi_34/d/%b1%d2%b5%dc%a4%b5%a4%f3%a4%c1%a4%ce%ba%a3%c6%fc%a4%ce%a4%b4%a4%cf%a4%f3</span></a>
```

**Comment by kmahelona** [comment]
I have a similar problem. When I have URLs that have diacritics like `ā` I get,
`https://cdn.tehiku.nz/2020/07/17/443896_Hauk%C4%81inga__Flensing_mp4_versions/playlist.m3u8`

However, given support for these types of unicode characters in URLs, I'd expect,
`https://cdn.tehiku.nz/2020/07/17/443896_Haukāinga__Flensing_mp4_versions/playlist.m3u8`

I notice that in the django admin my url's are displaying with the `ā` so perhaps it's DRF that's encoding urls a certain way and how can we change that? Can we make a universal option so that when we make a serializer with the `fields = "__all__"` we have an easy way to ensure any url related fields are encoded accordingly.

Specifically in my case, those URLs are coming from Django Storages - so they are file fields that have an s3 URL.

**Comment by tomchristie** [comment]
Behaviour likely resolved via #7548
